pywr/recorders/recorders.py

```diff
@@ -110,7 +110,29 @@ def setup(self):
         if self.nodes is None:
             self.node_names = sorted(self.model.nodes.keys())
         else:
-            self.node_names = sorted(n.name for n in self.nodes)
+            # the next line is commented because when the nodes names are provided (e.g. loaded from the json)
+            # it will throw an exception
+            # self.node_names = sorted(n.name for n in self.nodes)
+            nodes = []
+            node_names = []
+            for node_ in self.nodes:
+                # test if the node name is provided
+                if isinstance(node_, basestring):
+                    try:
+                        # lookup node by name
+                        nodes.append(self.model.nodes[node_])
+                        node_names.append(node_)
+                    except:
+                        raise ValueError ("node name is not in the model: {}".format(node_))
+                else:
+                    try:
+                        nodes.append(node_)
+                        node_names.append((node_.name))
+                    except:
+                        # raise an error if the object is not recognized
+                        raise ValueError("Unrecognised node object: {}".format(node_))
+            self.node_names = node_names
+            self.nodes = nodes
```

Comment:
I'm not sure I like updating `self.nodes`. If you pass a list of strings to `__int__` then `self.nodes` is that list of strings. However after `setup` is called it is now a new list of `Node` objects. For me this is odd that this changes. It appears the remainder of the code on relies on `self.node_names` so we only need to set that here.

Reply:
I agree, there is no need to update the list as it will not be used after that